go/tao/auth/parser.go

```diff
@@ -345,13 +345,13 @@ func (p *parser) expectTerm() (Term, error) {
 		if !ok {
 			return nil, fmt.Errorf("a keyword must be a string")
 		}
-		switch s {
-		case "ext":
+		switch {
+		case p.cur() == tokenExt:
 			return p.expectPrinTail()
-		case "key", "tpm":
+		case isPrinToken(p.cur()):
 			return p.expectPrin()
 		default:
-			return nil, fmt.Errorf(`expected "key", "tpm", or "ext", found %s`, s)
+			return nil, fmt.Errorf(`expected keyword of principal token, found %s`, s)
```

Comment:
or ext, which is a keyword rather than a principal token. You probably need to make sure that "ext" doesn't get defined as a principal token, since that will break everything in surprising ways.